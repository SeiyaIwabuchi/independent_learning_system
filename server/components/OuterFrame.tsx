import { AppBar, Toolbar, Typography, Snackbar } from "@material-ui/core";
import React, { ReactNode } from "react";



interface IProps {
    appbar : {
        title : string,
        leftButton? : ReactNode,
        rightButton? : ReactNode
    },
    snackbar : {
        state? : boolean,
        setState? : React.Dispatch<React.SetStateAction<boolean>>,
        msg? : string
    },
    children:ReactNode
}

const OuterFrame = (props: IProps) => {
    return (
        <div style={{maxWidth:"50%",margin:"0 auto"}}>
            <AppBar position="fixed">
                <Toolbar>
                    {props.appbar.leftButton}
                    <Typography variant="h6" style={{marginLeft:"10px"}}>
                        {props.appbar.title}
                    </Typography>
                    <div style={{marginLeft:"auto",marginRight:"0px"}}>{props.appbar.rightButton}</div>
                </Toolbar>
            </AppBar>
            <div style={{
                marginTop:"80px",
                marginRight:"10px",
                marginLeft : "10px"
                }}>
                {props.children}
            </div>
            <Snackbar
                anchorOrigin={{ vertical: "bottom", horizontal: "left" }}
                open={props.snackbar.state}
                onClose={ () => {
                    if(props.snackbar.setState != undefined) props.snackbar.setState(false)
                }}
                message={props.snackbar.msg}
                key={"bottomleft"}
            />
        </div>
        );
}
export default OuterFrame;