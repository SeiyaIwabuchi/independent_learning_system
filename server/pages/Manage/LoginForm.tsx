import { Typography, Button, InputLabel, OutlinedInput, FormControl } from "@material-ui/core";
import React, { useState } from "react";
import OuterFrame from "../../components/OuterFrame";
import router from "next/router";
import Form from "../../components/Form";
import LoginFormSchema_ from "../../form_schemas/LoginFormSchema.json";
import { LoginFormResponseShcema } from "../../form_schemas/ts/LoginFormShcema";
import { SessionId } from "../../query_param_shemas/SessionId";
import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos';
import { IconButton } from "@material-ui/core";
import { useEffect } from "react";
import { GetServerSideProps } from "next";

const LoginFormSchema: any = LoginFormSchema_.definitions.LoginFormShcema;

export const getServerSideProps: GetServerSideProps = async (context) => {
    return {
        props:{
            preSessionId : context.req.cookies.sessionId || "Unauthorised"
        }
    }
};

let LoginForm = (props : {preSessionId : string}) => {
    const appbar = {
        title: "ログイン",
        rightButton: <></>,
        leftButton: <IconButton onClick={()=> router.back()}><ArrowBackIosIcon/></IconButton>
    }
    const [snackbarState, setSnackbarState] = useState(false);
    const [snackbarMsg, setSnackbarMsg] = useState("");
    const snackbar = {
        state: snackbarState,
        setState: setSnackbarState,
        msg: snackbarMsg
    };
    useEffect( () => {
        // localStorageにsessionIdがある状態でログインフォームアクセスしたとき、
        //sessionIdが有効な時はそのまま管理トップに遷移する。
        if(props.preSessionId != "Unauthorised"){
            router.push(`/Manage/Top?sessionId=${props.preSessionId}`);
        }
    });
    return (
        <OuterFrame appbar={appbar} snackbar={snackbar}>
            <div style={{
                border : "2px solid #b3b1b1",
                borderRadius : "10px",
                padding : "20px"
            }}>
                <Form
                    action="/api/Login"
                    schema={LoginFormSchema}
                    submitButtonName="ログイン"
                    dataDest="/api/Login"
                    onApiRes={(json: LoginFormResponseShcema) => {
                        if (json.sessionId != "Unauthorised") {
                            router.push(`/Manage/SetCookie?${SessionId(json.sessionId)}&goto=${"/Manage/Top"}`);
                        } else {
                            setSnackbarMsg("ログインに失敗しました。ユーザ名またはパスワードが間違っています。");
                            setSnackbarState(true);
                        }
                    }}
                    onApiError={(err) => {
                        console.log(err);
                    }}
                    uiSchema={{
                        "password" : {
                            "ui:widget": "password"
                        }
                    }}
                    autoComplete="on"
                    name = "loginForm"
                    id = "loginForm"
                />
            </div>
        </OuterFrame>
    )
};

export default LoginForm;