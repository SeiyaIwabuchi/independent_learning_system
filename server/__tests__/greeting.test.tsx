/**
 * @jest-environment jsdom
 */

import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import renderer from 'react-test-renderer';
import OuterFrame from '../components/OuterFrame';
import LoginForm from '../pages/Manage/LoginForm';

test("greeting",()=>{
    const cmpt = render(<LoginForm preSessionId="1234567890"/>);
    console.log(cmpt.getByText(/ログイン/));
});

export default "this is test code.";