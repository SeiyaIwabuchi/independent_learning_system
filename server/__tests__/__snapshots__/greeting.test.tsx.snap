// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`greeting 1`] = `
<div
  style={
    Object {
      "margin": "0 auto",
      "maxWidth": "400px",
    }
  }
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        test
      </h6>
      <div
        style={
          Object {
            "marginLeft": "auto",
            "marginRight": "0px",
          }
        }
      />
    </div>
  </header>
  <div
    style={
      Object {
        "marginLeft": "10px",
        "marginRight": "10px",
        "marginTop": "80px",
      }
    }
  />
</div>
`;
